src/main/java/com/hivemq/configuration/service/InternalConfigurations.java

```diff
@@ -469,4 +470,5 @@ public class InternalConfigurations {
     public static boolean EXPIRE_INFLIGHT_MESSAGES_ENABLED = false;
     public static boolean EXPIRE_INFLIGHT_PUBRELS_ENABLED = false;
```

Comment:
Imo, we should think about not introducing this feature/bugfix and even remove the expiry pubrel behavior from CE as it breaks the QoS-2 flow.

Reply:
I consider it currently as a bug which should be fixed. Get your point why we should remove it, can you create a ticket for that?

Reply:
I actually think that it should be ok to leave it as-is even. This should be the conscious decision of the user to set this flag to true to 'adapt' the QoS2 flow. In any case, we cannot do anything about it - CE is open-source, so people could fork it and add this functionality anyway. Also, the default MAXIMUM_INFLIGHT_PUBREL_EXPIRY is also quite high. 

Overall, I'm fine with either option but IMHO this is a separate issue. For now, I suggest to add a warning to the internal config. What do you think?